http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GceRuntimeMonitorInterp.scala

```diff
@@ -0,0 +1,698 @@
+    for {
+      now <- nowInstant
+      implicit0(ct: ApplicativeAsk[F, AppContext]) = ApplicativeAsk.const[F, AppContext](
+        AppContext(monitorContext.traceId, now)
+      )
+      r <- gcsMonitorState match {
+        case GceMonitorState.Initial => handleInitial(monitorContext)
+        case GceMonitorState.CheckTools(ip, runtimeAndRuntimeConfig, images) =>
+          handleCheckTools(monitorContext, runtimeAndRuntimeConfig, ip, images)
+        case GceMonitorState.Check(runtimeAndRuntimeConfig) => handleCheck(monitorContext, runtimeAndRuntimeConfig)
+      }
+    } yield r
+
+  private[monitor] def handleInitial(
+    monitorContext: MonitorContext
+  )(implicit ct: ApplicativeAsk[F, AppContext]): F[CheckResult] =
+    for {
+      runtimeAndRuntimeConfig <- getDbRuntimeAndRuntimeConfig(monitorContext.runtimeId)
+      next <- runtimeAndRuntimeConfig.runtime.status match {
+        case status if status.isMonitored =>
+          logger.info(
+            s"${monitorContext} | Start monitor runtime ${runtimeAndRuntimeConfig.runtime.projectNameString}'s ${status} process."
+          ) >> handleCheck(monitorContext, runtimeAndRuntimeConfig)
+        case _ =>
+          F.pure(((), None): CheckResult)
+      }
+    } yield next
+
+  /**
+   * Queries Google for the cluster status and takes appropriate action depending on the result.
+   * @return ClusterMonitorMessage
+   */
+  private def handleCheck(monitorContext: MonitorContext, runtimeAndRuntimeConfig: RuntimeAndRuntimeConfig)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[CheckResult] =
+    for {
+      instance <- googleComputeService.getInstance(
+        runtimeAndRuntimeConfig.runtime.googleProject,
+        config.gceZoneName,
+        InstanceName(runtimeAndRuntimeConfig.runtime.runtimeName.asString)
+      )
+      result <- runtimeAndRuntimeConfig.runtime.status match {
+        case RuntimeStatus.Creating =>
+          creatingRuntime(instance, monitorContext, runtimeAndRuntimeConfig)
+        // This is needed because during boot time, we no longer have reference to the previous delete operation
+        // But this path should only happen during boot and there's on-going delete
+        case RuntimeStatus.Deleting =>
+          instance match {
+            case None => deletedRuntime(monitorContext, runtimeAndRuntimeConfig)
+            case _    => checkAgain(monitorContext, runtimeAndRuntimeConfig, Some("Instance hasn't been deleted yet"))
+          }
+        case RuntimeStatus.Starting =>
+          startingRuntime(instance, monitorContext, runtimeAndRuntimeConfig)
+        // This is needed because during boot time, we no longer have reference to the previous delete operation
+        // But this path should only happen during boot and there's on-going delete
+        case RuntimeStatus.Stopping =>
+          val gceStatus = instance.flatMap(i => GceInstanceStatus.withNameInsensitiveOption(i.getStatus))
```

Comment:
Wondering if the logic in this case could be put into a `stopRuntime` method for consistency sake

Reply:
`stopRuntime` is called in `pollCheck` as well, and the additional code won't make sense in the case of `pollCheck`...we can potentially have another function wrapping this code, but I kinda prefer to have less additional functions if this code doesn't grow further.